mailet/standard/src/test/java/org/apache/james/transport/mailets/HeadersToHTTPTest.java

```diff
@@ -0,0 +1,201 @@
+    private static UriHttpRequestHandlerMapper mapper;
+    private Mail mail;
+
+    private String urlTestPattern;
+
+    @BeforeAll
+    public static void setupServer() throws MessagingException, IOException {
+        mapper = new UriHttpRequestHandlerMapper();
+
+        SocketConfig socketConfig = SocketConfig.custom().setSoTimeout(50000).build();
+        server = ServerBootstrap.bootstrap().setListenerPort(0).setSocketConfig(socketConfig)
+                .setExceptionLogger(ExceptionLogger.NO_OP).setHandlerMapper(mapper).create();
+
+        server.start();
+
+    }
+
+    @AfterAll
+    public static void shutdown() {
+        server.shutdown(5L, TimeUnit.SECONDS);
+    }
+
+    @BeforeEach
+    public void setup() throws MessagingException, IOException {
+        mail = MailUtil.createMockMail2Recipients(MimeMessageUtil.mimeMessageFromStream(
+                ClassLoader.getSystemResourceAsStream("mime/sendToRemoteHttp.mime")));
+    }
+
+    @AfterEach
+    public void cleanMapper() throws MessagingException, IOException {
+        mapper.unregister(urlTestPattern);
+    }
+
+    @Test
+    public void shouldBeFailedWhenServiceNotExists() throws Exception {
+
+        urlTestPattern = "/path/to/service/failed";
+
+        FakeMailetConfig mailetConfig = FakeMailetConfig.builder()
+                .setProperty("parameterKey", "pKey").setProperty("parameterValue", "pValue")
+                .setProperty("url", "http://qwerty.localhost:123456" + urlTestPattern).build();
+
+        Mailet mailet = new HeadersToHTTP();
+        mailet.init(mailetConfig);
+
+        mailet.service(mail);
+
+        assertThat(mail.getMessage().getHeader("X-headerToHTTP")).hasSize(1)
+                .allSatisfy((header) -> assertThat(header).isEqualTo("Failed"));
+        assertThat(mail.getMessage().getHeader("X-headerToHTTPFailure")).hasSize(1)
+                .allSatisfy((header) -> assertThat(header).isNotBlank());
+
+    }
+
+    @Test
+    public void shouldBeSucceededWhenServiceResponseIsOk() throws Exception {
+
+        urlTestPattern = "/path/to/service/succeeded";
+
+        FakeMailetConfig mailetConfig = FakeMailetConfig.builder()
+                .setProperty("parameterKey", "pKey").setProperty("parameterValue", "pValue")
+                .setProperty("url", "http://" + server.getInetAddress().getHostAddress() + ":"
+                        + server.getLocalPort() + urlTestPattern)
+                .build();
+
+        mapper.register(urlTestPattern, (request, response, context) -> {
+
+            assertThat(request.getRequestLine().getMethod()).isEqualTo("POST");
+
+            BasicHttpEntityEnclosingRequest basicRequest = (BasicHttpEntityEnclosingRequest) request;
+            BasicHttpEntity entity = (BasicHttpEntity) basicRequest.getEntity();
+
+            try {
+                List<NameValuePair> params = URLEncodedUtils.parse(entity);
+                assertThat(params).hasSize(5).anySatisfy((param) -> {

```

Comment:
these assertions belong to another test in my opinion (it doesn't match "shoud succeed" description)